app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Callable:
+    memo = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        key = args + tuple(sorted(kwargs.items()))
+        if key in memo:
+            print("Getting from cache")
+            return memo[key]
+        else:
```

Comment:
else is useless here, because you have 3 returns

Reply:
changed